submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestValidator.kt

```diff
@@ -0,0 +1,66 @@
+package ac.uk.ebi.biostd.submission.domain.request
+
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionRequest
+import ac.uk.ebi.biostd.persistence.common.service.RqtUpdate
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestPersistenceService
+import ebi.ac.uk.base.orFalse
+import ebi.ac.uk.model.RequestStatus
+import ebi.ac.uk.model.RequestStatus.INVALID
+import ebi.ac.uk.model.RequestStatus.VALIDATED
+import ebi.ac.uk.security.integration.components.IUserPrivilegesService
+import mu.KotlinLogging
+import uk.ac.ebi.events.service.EventsPublisherService
+
+private val logger = KotlinLogging.logger {}
+
+class SubmissionRequestValidator(
+    private val userPrivilegesService: IUserPrivilegesService,
+    private val eventsPublisherService: EventsPublisherService,
+    private val queryService: SubmissionPersistenceQueryService,
+    private val requestService: SubmissionRequestPersistenceService,
+) {
+    suspend fun validateRequest(
+        accNo: String,
+        version: Int,
+        processId: String,
+    ) {
+        var requestStatus = INVALID
+
+        requestService.onRequest(accNo, version, RequestStatus.INDEXED_CLEANED, processId) {
+            requestStatus = validateRequest(it)
+            RqtUpdate(it.withNewStatus(requestStatus))
+        }
+
+        if (requestStatus == VALIDATED) eventsPublisherService.requestValidated(accNo, version)
+    }
+
+    internal suspend fun validateRequest(rqt: SubmissionRequest): RequestStatus {
```

Comment:
smalI detail but I do not see any value in this method to be honest,  single line that call another method with basically doing anything

Reply:
Yeah, I scratched my head with this one for a while. I left it there because I wanted to leave the template for the future when we plug in the other validations but it's not required now and we can refactor when we actually need it. I'll remove it 👍